src/compiler/test-file/formats/typescript.js

```diff
@@ -36,29 +41,46 @@ export default class TypeScriptTestFileCompiler extends APIBasedTestFileCompiler
         throw new Error(errMsg);
     }
 
+    static _normalizeFilename (filename) {
+        filename = path.resolve(filename);
+
+        if (OS.win)
+            filename = filename.toLowerCase();
+
+        return filename;
+    }
+
     _compileCode (code, filename) {
         // NOTE: lazy load the compiler
         var ts = require('typescript');
 
+        filename = TypeScriptTestFileCompiler._normalizeFilename(filename);
+
         if (this.cache[filename])
             return this.cache[filename];
 
-        var opts        = TypeScriptTestFileCompiler._getTypescriptOptions();
-        var program     = ts.createProgram([filename], opts);
+        var opts    = TypeScriptTestFileCompiler._getTypescriptOptions();
+        var program = ts.createProgram([filename], opts);
+
+        program.getSourceFiles().forEach(sourceFile => {
+            sourceFile.renamedDependencies = RENAMED_DEPENDENCIES_MAP;
+        });
+
         var diagnostics = ts.getPreEmitDiagnostics(program);
 
         if (diagnostics.length)
             TypeScriptTestFileCompiler._reportErrors(diagnostics);
 
-        var result = ts.transpileModule(code, {
-            compilerOptions:     opts,
-            fileName:            filename,
-            renamedDependencies: { testcafe: APIBasedTestFileCompilerBase.EXPORTABLE_LIB_PATH }
-        });
+        // NOTE: First argument of emit() is a source file for compilation. If it's undefined, all files in <program> will be compiled.
+        // <program> contains the file specified in createProgram() plus all its dependencies. This mode is a lot faster than
+        // compiling files one-by-one, and used in tsc CLI compiler.
```

Comment:
it's an object created by the `ts.createProgram([filename], opts);` execution (see above)